### ISSUE TITLE
Compile error when 'path/idris/mingw' has spaces; mingw-w64 substitute works

### ISSUE THREAD

**Issue by GezzMC** [idris-lang/Idris-dev#3437]
Hi, thanks for the product. Looks promising.

I work with spaces in my paths for portability. When spaces are in the Idris tool path, the command `idris --execute hello.idr` works, but not `idris hello.idr -o hello.exe`.

I include the complete error message below, but it's because I have a space in the folder `d ev`.

I substituted a MinGW-w64 from the following page, and it works, at least for my test here:

[sourceforge.net/projects/mingw-w64, v5.2.0, posix threads, seh](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/5.2.0/threads-posix/seh/)

Thanks. Here's the complete error message:

```
E:\E_1\02-p\3+\idr\(_jv wrk_)\samples__tutorial>idris hello.idr -o hello.exe 
E:/E_2/d ev/xbin/idris-0.12.2/mingw/bin/../lib/gcc/x86_64-w64-mingw32/5.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find E:/E_2/d: No such file or directory
E:/E_2/d ev/xbin/idris-0.12.2/mingw/bin/../lib/gcc/x86_64-w64-mingw32/5.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find ev/xbin/idris-0.12.2/mingw/bin/../lib/gcc/x86_64-w64-mingw32/5.2.0/../../../../x86_64-w64-mingw32/lib/../lib/default-manifest.o: No such file or directory
collect2.exe: error: ld returned 1 exit status
FAILURE: "gcc" ["-O2","-fwrapv","-fno-strict-overflow","-DHAS_PTHREAD","-DIDRIS_ENABLE_STATS","-I.","C:\\Users\\jv\\AppData\\Local\\Temp\\idris1477111538.c","-LE:\\E_2\\d ev\\xbin\\idris-0.12.2\\./libs\\rts","-lidris_rts","-lpthread","-IE:\\E_2\\d ev\\xbin\\idris-0.12.2\\./libs\\rts","-lm","-I.","-IE:\\E_2\\d ev\\xbin\\idris-0.12.2\\./libs\\prelude","-IE:\\E_2\\d ev\\xbin\\idris-0.12.2\\./libs\\base","-IE:\\D_1\\w rk\\g_d\\gezz\\xbin\\idris-0.12.2\\./libs\\prelude","-IE:\\D_1\\w rk\\g_d\\gezz\\xbin\\idris-0.12.2\\./libs\\base","-ID:\\Niklas\\repo\\Idris-dev\\dist\\dist-sandbox-8527c5fc\\build\\idris\\./libs\\prelude","-ID:\\Niklas\\repo\\Idris-dev\\dist\\dist-sandbox-8527c5fc\\build\\idris\\./libs\\base","-I.","-ID:\\Niklas\\repo\\Idris-dev\\.cabal-sandbox\\x86_64-windows-ghc-7.10.3\\idris-0.12.2\\prelude","-ID:\\Niklas\\repo\\Idris-dev\\.cabal-sandbox\\x86_64-windows-ghc-7.10.3\\idris-0.12.2\\base","-Wl,--stack,16777216","-o","hello.exe"]
```

**Comment by melted** [comment]
Thanks for the report, I'll look into it. But  it does look like a problem with the gcc as the arg list that was printed had no wrongly split paths.

**Comment by davidgrenier** [comment]
Moving the Idris folder to one without spaces solved the problem for me as well.

**Comment by melted** [comment]
The solution to this is that I will bundle a newer gcc when I build the
next Idris release.

Den 16 nov. 2016 17:13 skrev "David Grenier" <notifications@github.com>:

**Comment by melted** [comment]
Should be fixed with the 0.99 release with an upgraded gcc.
